### PR TITLE
Update MultiInstanceConnecting

### LINKED ISSUE #394: Add shuffle option for addresses
It's needed when you use a round-robin connection chooser with many instances of cartridge-java application

### PR DESCRIPTION
**Pull request by iDneprov**
Added an explanation of when addresses need to be shuffled

I haven't forgotten about:
- [x] Tests
- [ ] Changelog
- [x] Documentation
- [x] Commit messages comply with the [guideline](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message)
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:

Closes #394

### CODE REVIEW

File: docs/MultiInstanceConnecting.md [hunk 1]
```diff
@@ -5,19 +5,24 @@
 This section shows examples of how to connect to multiple [Cartridge](https://github.com/tarantool/cartridge) routers.  
 We recommend to use balancer in production applications rather than leaving this task to the connector.
 
+If you are using **multiple tarantool clients** do not forget to **shuffle addresses** before providing them to clients.
+TarantoolClient uses [round-robin](https://en.wikipedia.org/wiki/Round-robin_scheduling) so without URIs shuffle
+the first request of each of your clients will be sent to the same instance.
```
Comment (dkasimovskiy) [inline docs/MultiInstanceConnecting.md:10]:
```suggestion
Without addresses **shuffle** the first request of each of your clients will be sent to the same instance.

**Review by dkasimovskiy — CHANGES_REQUESTED** [review]

File: docs/MultiInstanceConnecting.md [hunk 2]
```diff
@@ -5,19 +5,24 @@
 This section shows examples of how to connect to multiple [Cartridge](https://github.com/tarantool/cartridge) routers.  
 We recommend to use balancer in production applications rather than leaving this task to the connector.
 
+If you are using **multiple tarantool clients** do not forget to **shuffle addresses** before providing them to clients.
+TarantoolClient uses [round-robin](https://en.wikipedia.org/wiki/Round-robin_scheduling) so without URIs shuffle
```
[the remark this answers is not in the document]
Comment (ArtDu) [reply docs/MultiInstanceConnecting.md]:
```suggestion
By default TarantoolClient uses [round-robin](https://en.wikipedia.org/wiki/Round-robin_scheduling) approach to separate load between Tarantool instances.

```

File: docs/MultiInstanceConnecting.md (hunk 1 above)
Comment (ArtDu) [reply docs/MultiInstanceConnecting.md]:
```suggestion
> **Warning**
> Without addresses **shuffle** you may have a situation when clients will send something to the same instances. 
> Especially when you start your clients in parallel at the same time.
```

File: docs/MultiInstanceConnecting.md [hunk 3]
```diff
@@ -3,21 +3,26 @@
 # Connecting to multiple Tarantool instances
 
 This section shows examples of how to connect to multiple [Cartridge](https://github.com/tarantool/cartridge) routers.  
-We recommend to use balancer in production applications rather than leaving this task to the connector.
+We recommend to use load balancer in production applications rather than leaving this task to the connector.
+
+In case of using **multiple instances of TarantoolClient** we strongly recommend to **shuffle** list of addresses before providing them to clients.
```
Comment (akudiyar) [inline docs/MultiInstanceConnecting.md:8]:
```suggestion
In the case of using **multiple instances** of `TarantoolClient` we strongly recommend to apply a **shuffle** function to the list of addresses.
```

File: docs/MultiInstanceConnecting.md [hunk 4]
```diff
@@ -3,21 +3,26 @@
...
+By default TarantoolClient uses [round-robin](https://en.wikipedia.org/wiki/Round-robin_scheduling) approache to separate load between Tarantool instances.
```
Comment (akudiyar) [inline docs/MultiInstanceConnecting.md:9]:
```suggestion
By default, `TarantoolClient` uses the [round-robin](https://en.wikipedia.org/wiki/Round-robin_scheduling) approach for selecting the next available connection to distribute the load between Tarantool instances.
```

File: docs/MultiInstanceConnecting.md [hunk 5]
```diff
@@ -3,21 +3,26 @@
...
+Without addresses **shuffle** the first request of each of your clients will be sent to the same instance.
```
Comment (akudiyar) [inline docs/MultiInstanceConnecting.md:10]:
```suggestion
When several instances of `TarantoolClient` are used simultaneously, with the default connection selection strategy and without **shuffling** of the server addresses the first request from every instance will be sent to the same server.
```

File: docs/MultiInstanceConnecting.md [hunk 6]
```diff
@@ -3,21 +3,26 @@
...
+You do not have to worry about shuffling addresses if you are using single TarantoolClient connected to multiple instances.
```
Comment (akudiyar) [inline docs/MultiInstanceConnecting.md:11]:
```suggestion
You do not have to worry about shuffling addresses if you are using a single `TarantoolClient` connected to multiple instances, or if you are using a custom connection selection strategy that takes into account the problem of simultaneous requests.
```

File: docs/MultiInstanceConnecting.md (hunk 5 above)
Comment (akudiyar) [reply docs/MultiInstanceConnecting.md:10]: What problem do the simultaneous requests cause? We need to add a line about that, currently the readers will be confused

**Review by akudiyar — CHANGES_REQUESTED** [review]
Some adjustments for the text

**Comment by iDneprov** [comment]
@dkasimovskiy Ready for review. Fixed by @akudiyar comments.

**Review by dkasimovskiy — APPROVED** [review]

**Pull request merged**